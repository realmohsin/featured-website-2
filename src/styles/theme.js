import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
