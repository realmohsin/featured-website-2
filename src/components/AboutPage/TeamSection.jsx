import React from 'react'
import { graphql, useStaticQuery } from 'gatsby'
import { GatsbyImage, getImage } from 'gatsby-plugin-image'
import { makeStyles } from '@material-ui/core/styles'
import { Container, Grid, Typography } from '@material-ui/core'
import teamMembers from '../../data/team-members'

const useStyles = makeStyles(theme => ({
  teamSection: {
    padding: '4rem 0 10rem',
    [theme.breakpoints.down('xs')]: {
      padding: '4rem 0 6rem'
    }
  },
  teamSectionTitle: {
    ...theme.custom.title2,
    textAlign: 'left',
    paddingLeft: '5rem',
    fontSize: '5.6rem',
    [theme.breakpoints.down('md')]: {
      fontSize: '4.6rem'
    },
    [theme.breakpoints.down('xs')]: {
      fontSize: '3.4rem',
      paddingLeft: '3.5rem',
      marginBottom: '2rem'
    }
  },
  gridItem: {
    padding: '2.5rem',
    [theme.breakpoints.down('xs')]: {
      padding: '2.2rem 3.5rem'
    }
  },
  name: {
    ...theme.custom.title5,
    letterSpacing: '0.7px',
    marginTop: '1rem',
    [theme.breakpoints.down('xs')]: {
      marginTop: 0
    }
  },
  title: {
    // textTransform: 'uppercase',
    fontSize: '1.4rem'
  }
}))

const TeamSection = props => {
  const classes = useStyles()
  const data = useStaticQuery(graphql`
    query {
      allFile(filter: { name: { regex: "/^legal-team/" } }) {
        nodes {
          childImageSharp {
            gatsbyImageData
          }
        }
      }
    }
  `)

  return (
    <section className={classes.teamSection}>
      <Container>
        <h2 className={classes.teamSectionTitle}>Legal Team</h2>
        <Grid container>
          {teamMembers.map(({ id, name, title }, i) => (
            <Grid key={id} item xs={12} sm={6} md={4} className={classes.gridItem}>
              <figure>
                <GatsbyImage
                  image={getImage(data.allFile.nodes[i].childImageSharp)}
                  alt='Business Law'
                />
                <figcaption>
                  <Typography
                    variant='body1'
                    className={classes.name}
                  >
                    {name}
                  </Typography>
                  <Typography
                    variant='body1'
                    className={classes.title}
                  >
                    {title}
                  </Typography>
                </figcaption>
              </figure>
            </Grid>
          ))}
        </Grid>
      </Container>
    </section>
  )
}

export default TeamSection
