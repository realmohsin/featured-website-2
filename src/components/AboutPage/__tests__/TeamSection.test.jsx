import React from "react";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import TeamSection from "../TeamSection";

// Mock Gatsby's useStaticQuery and graphql
jest.mock("gatsby", () => ({
  useStaticQuery: jest.fn(() => ({
    allFile: {
      nodes: [
        {
          childImageSharp: {
            gatsbyImageData: {
              images: {
                fallback: {
                  src: "/static/team-member-1.jpg",
                },
              },
            },
          },
        },
        {
          childImageSharp: {
            gatsbyImageData: {
              images: {
                fallback: {
                  src: "/static/team-member-2.jpg",
                },
              },
            },
          },
        },
        {
          childImageSharp: {
            gatsbyImageData: {
              images: {
                fallback: {
                  src: "/static/team-member-3.jpg",
                },
              },
            },
          },
        },
      ],
    },
  })),
  graphql: jest.fn(),
}));

// Mock gatsby-plugin-image
jest.mock("gatsby-plugin-image", () => ({
  GatsbyImage: ({ image, alt, ...props }) => (
    <img
      src={image?.images?.fallback?.src || "/mock-team-image.jpg"}
      alt={alt}
      {...props}
    />
  ),
  getImage: jest.fn((image) => image),
}));

// Mock team members data
jest.mock("../../../data/team-members", () => [
  {
    id: 1,
    name: "Domenic Romano",
    title: "Managing Partner",
  },
  {
    id: 2,
    name: "Sarah Johnson",
    title: "Senior Associate",
  },
  {
    id: 3,
    name: "Michael Chen",
    title: "Legal Counsel",
  },
]);

const theme = createMuiTheme({
  custom: {
    title2: {
      fontSize: "5.6rem",
      fontWeight: "bold",
    },
    title5: {
      fontSize: "1.8rem",
    },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("TeamSection Component", () => {
  it("renders team section with all members", () => {
    const { container } = renderWithTheme(<TeamSection />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("displays section title", () => {
    renderWithTheme(<TeamSection />);
    expect(screen.getByText(/legal team/i)).toBeInTheDocument();
  });

  it("renders all team members", () => {
    renderWithTheme(<TeamSection />);
    
    expect(screen.getByText(/domenic romano/i)).toBeInTheDocument();
    expect(screen.getByText(/sarah johnson/i)).toBeInTheDocument();
    expect(screen.getByText(/michael chen/i)).toBeInTheDocument();
  });

  it("displays team member titles", () => {
    renderWithTheme(<TeamSection />);
    
    expect(screen.getByText(/managing partner/i)).toBeInTheDocument();
    expect(screen.getByText(/senior associate/i)).toBeInTheDocument();
    expect(screen.getByText(/legal counsel/i)).toBeInTheDocument();
  });

  it("renders team member images", () => {
    renderWithTheme(<TeamSection />);
    
    const images = screen.getAllByRole("img");
    expect(images).toHaveLength(3);
    
    images.forEach((img) => {
      expect(img).toHaveAttribute("alt", "Business Law");
    });
  });

  it("uses proper semantic structure with figures and figcaptions", () => {
    renderWithTheme(<TeamSection />);
    
    const figures = screen.getAllByRole("figure");
    expect(figures).toHaveLength(3);
  });
});
