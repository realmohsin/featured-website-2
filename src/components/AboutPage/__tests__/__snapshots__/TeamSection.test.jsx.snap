// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamSection Component renders team section with all members 1`] = `
<section
  class="makeStyles-teamSection-1"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <h2
      class="makeStyles-teamSectionTitle-2"
    >
      Legal Team
    </h2>
    <div
      class="MuiGrid-root MuiGrid-container"
    >
      <div
        class="MuiGrid-root makeStyles-gridItem-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <figure>
          <img
            alt="Business Law"
            src="/mock-team-image.jpg"
          />
          <figcaption>
            <p
              class="MuiTypography-root makeStyles-name-4 MuiTypography-body1"
            >
              Domenic Romano
            </p>
            <p
              class="MuiTypography-root makeStyles-title-5 MuiTypography-body1"
            >
              Managing Partner
            </p>
          </figcaption>
        </figure>
      </div>
      <div
        class="MuiGrid-root makeStyles-gridItem-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <figure>
          <img
            alt="Business Law"
            src="/mock-team-image.jpg"
          />
          <figcaption>
            <p
              class="MuiTypography-root makeStyles-name-4 MuiTypography-body1"
            >
              Sarah Johnson
            </p>
            <p
              class="MuiTypography-root makeStyles-title-5 MuiTypography-body1"
            >
              Senior Associate
            </p>
          </figcaption>
        </figure>
      </div>
      <div
        class="MuiGrid-root makeStyles-gridItem-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4"
      >
        <figure>
          <img
            alt="Business Law"
            src="/mock-team-image.jpg"
          />
          <figcaption>
            <p
              class="MuiTypography-root makeStyles-name-4 MuiTypography-body1"
            >
              Michael Chen
            </p>
            <p
              class="MuiTypography-root makeStyles-title-5 MuiTypography-body1"
            >
              Legal Counsel
            </p>
          </figcaption>
        </figure>
      </div>
    </div>
  </div>
</section>
`;
